impl/src/main/java/org/apache/myfaces/renderkit/html/HtmlResponseStateManager.java

```diff
@@ -121,6 +121,9 @@ private void writeViewStateField(FacesContext facesContext, ResponseWriter respo
     {
         String serializedState = getStateCache(facesContext).getStateTokenProcessor(facesContext)
                 .encode(facesContext, savedState);
+
+        log.info("Writing serialized viewstate string: " + serializedState);
```

Comment:
This will always concat the string shouldn't you wrap this in something like...

```
if (log.isInfoEnabled()) { log.info() )} 
```

Reply:
Good point!